en/companion_computer/holybro_pixhawk_jetson_baseboard.md

```diff
@@ -0,0 +1,1177 @@
+```
+
+## ROS 2 Setup
+
+[ROS 2](../ros2/index.md) is powerful robotics API that you can run on the Jetson in order to control PX4 running on the Pixhawk.
+It is harder to learn than MAVLink and its interfaces still evolving, but it offers a much deeper integration with PX4, and much lower latency communication.
+
+In order to use ROS 2 we first need to run [XRCE-DDS middleware](../middleware/uxrce_dds.md) client on PX4 and agent on the Jetson.
+The instructions depend on the communications channel used (Ethernet or Serial).
+
+This section explains the specific setup, and is heavily based on the [ROS 2 user guide](../ros2/user_guide.md).
+
+### PX4 XRCE-DDS Client Setup
+
+The uXRCE-DDS client is built into PX4 firmware by default, and exposes a predefined set of uORB topics to the DDS network on which ROS 2 runs, via the XRCE-Agent.
+The set of exposed topics are defined in a build configuration file.
+
+The channel used by the client for communication with the agent is configured using parameters.
+You can use either the internal serial connection or the internal Ethernet switch.
+
+#### Serial Port Setup
+
+As covered in the MAVLink section, there is an internal serial link from Pixhawk `TELEM2` to Jetson `THS1`, which is configured to communicate via MAVLink by default.
+You will need to disable the MAVLink instance and enable the UXRCE_DDS configuration.
+
+You can [modify the parameters](../advanced_config/parameters.md) in QGroundControl parameter editor, or using `param set` in the [mavlink_shell](../debug/mavlink_shell.md).
+The code below assumes you're commands in the MAVLink shell.
+
+```sh
+param set MAV_1_CONFIG 0  # Disable MAVLINK on TELEM2 (so it can be used for XRCE-DDS)
+param set UXRCE_DDS_CFG 102 # Set UXRCE_DDS_CFG to TELEM2
+```
+
+::: info
+Note that you can also change the following parameters, but this requires a deeper understanding of DDS.
+
+```sh
+param set UXRCE_DDS_DOM_ID 0
+param set UXRCE_DDS_PTCFG 0
+param set UXRCE_DDS_SYNCC 0
+param set UXRCE_DDS_SYNCT 1
+```
+
+:::
+
+Then reboot the Pixhawk.
+
+#### Ethernet Setup
+
+As covered in the MAVLink section, there is an internal Ethernet switch linking the Jetson and Pixhawk.
+We configured both boards to use the same subnet above.
+However we need to disable MAVLink on the `Ethernet` port and enable `XRCE-DDS`.
+
+You can [modify the parameters](../advanced_config/parameters.md) in QGroundControl parameter editor, or using `param set` in the [MAVLink sheel](../debug/mavlink_shell.md).
+The code below assumes you're commands in the MAVLink shell.
+
+```sh
+param set MAV_2_CONFIG 0  # Disable MAVLINK on Ethernet (so Ethernet can be used for XRCE-DDS)
+param set UXRCE_DDS_CFG 1000 # Ethernet
+param set UXRCE_DDS_PRT 8888  # Set port to 8888
+```
+
+#### Checking the XRCE-DDS Client is Running
+
+::: tip
+The client should automatically start on boot!
+:::
+
+We can check that the client is running using the following command in the MAVLink shell:
```

Comment:
@farhangnaderi I can just connect USB-C to the pixhawk to set these from the shell in QGC right?